app/assets/javascripts/app/core/models/discussion.js

```diff
@@ -5,63 +5,59 @@
     .module('council.core')
     .factory('Discussion', Discussion);
 
-  function Discussion(DS, $http, $q) {
+  function Discussion(DiscussionComment, User, DS, $http, $q) {
     var ALLOWED_FIELDS = 'title subtitle body tags'.split(' ');
 
     var Discussion = DS.defineResource({
       name: 'discussion',
       endpoint: 'discussions',
+
       methods: {
         addComment: function(comment) {
-          var refresh = angular.bind(this, this.refresh);
+          var discussion = this;
 
-          return DS.create('discussionComment', comment)
-            .then(refresh);
+          return DS.create('discussionComment', comment, {
```

Comment:
It feels weird overriding the default options here in this model.
I'd have to check, but assuming `comment` has a `discussion_id` and you add `DS.linkInverse('discussionComment', comment.id)` in the `afterInject` in the *DiscussionComment* model, you could simply refactor this method into:
```
addComment: function(comment) {
  return DS.create('discussionComment', comment);
}
```